pkg/webservice/handlers_test.go

```diff
@@ -815,6 +824,131 @@ func addAndConfirmApplicationExists(t *testing.T, partitionName string, partitio
+	node2 := addNode(t, defaultPartition, "node-2", resources.NewResourceFromMap(map[string]resources.Quantity{"memory": 10, "vcore": 5}))
+	node3 := addNode(t, defaultPartition, "node-3", resources.NewResourceFromMap(map[string]resources.Quantity{"memory": 20, "vcore": 15}))
+	node4 := addNode(t, gpuPartition, "node-4", resources.NewResourceFromMap(map[string]resources.Quantity{"gpu": 10}))
+	// add allocatedResource to nodes
+	addAllocatedResource(t, node1, "alloc-1", "app-1", map[string]resources.Quantity{"memory": 1})
+	addAllocatedResource(t, node2, "alloc-1", "app-1", map[string]resources.Quantity{"memory": 1, "vcore": 1})
+	addAllocatedResource(t, node3, "alloc-1", "app-1", map[string]resources.Quantity{"memory": 1, "vcore": 1})
+	addAllocatedResource(t, node4, "alloc-1", "app-1", map[string]resources.Quantity{"gpu": 1})
+
+	// get nodes utilizations
+	getNodeUtilisations(resp, req)
+	err = json.Unmarshal(resp.outputBytes, &partitionNodesUtilDAOInfo)
+	assert.NilError(t, err, "should decode a list of *dao.PartitionNodesUtilDAOInfo")
+	assert.Equal(t, len(partitionNodesUtilDAOInfo), 2)
+	assert.Equal(t, partitionNodesUtilDAOInfo[0].ClusterId, rmID)
+
+	defaultPartitionNodesUtilDAOInfo := partitionNodesUtilDAOInfo[0]
+	gpuPartitionNodesUtilDAOInfo := partitionNodesUtilDAOInfo[1]
+	if defaultPartitionNodesUtilDAOInfo.Partition == "gpu" {
+		defaultPartitionNodesUtilDAOInfo = partitionNodesUtilDAOInfo[1]
+		gpuPartitionNodesUtilDAOInfo = partitionNodesUtilDAOInfo[0]
+	}
+
+	expectedResourceTypeCount := 2
+	assert.Equal(t, len(defaultPartitionNodesUtilDAOInfo.NodesUtilList), expectedResourceTypeCount)
+	expectedResourceTypeCount = 1
+	assert.Equal(t, len(gpuPartitionNodesUtilDAOInfo.NodesUtilList), expectedResourceTypeCount)
+
+	expectedNodeCount := 3
+	assertNodeUtilisationContent(t, defaultPartitionNodesUtilDAOInfo, "memory", expectedNodeCount)
+	expectedNodeCount = 2
+	assertNodeUtilisationContent(t, defaultPartitionNodesUtilDAOInfo, "vcore", expectedNodeCount)
+	expectedNodeCount = 1
+	assertNodeUtilisationContent(t, gpuPartitionNodesUtilDAOInfo, "gpu", expectedNodeCount)
+}
+
+func assertNodeUtilisationContent(t *testing.T, partitionNodesUtilDAOInfo *dao.PartitionNodesUtilDAOInfo, resourceType string, expectedNodeCount int) {
+	t.Helper()
+	nodeUtilisation := getNodesUtiliByType(t, partitionNodesUtilDAOInfo.NodesUtilList, resourceType)
+	assert.Equal(t, nodeUtilisation.ResourceType, resourceType, fmt.Sprintf("should have returned '%s', but got '%s'", resourceType, nodeUtilisation.ResourceType))
+	assert.Equal(t, len(nodeUtilisation.NodesUtil), 10, fmt.Sprintf("should have 10 bucket, but got %d", len(nodeUtilisation.NodesUtil)))
+	assert.Assert(t,
+		confirmNodeCount(nodeUtilisation.NodesUtil, int64(expectedNodeCount)),
+		fmt.Sprintf("unexpected number of nodes returned, should be %d", expectedNodeCount),
+	)
+}
+
+func getNodesUtiliByType(t *testing.T, nodesUtilList []*dao.NodesUtilDAOInfo, resourceType string) *dao.NodesUtilDAOInfo {
```

Comment:
`getNodesUtilisationByType`?

Reply:
I still prefer to use getNodesUtilByType() because the internal hierarchy of node_util.go is
1. Partition**NodesUtil**DAOInfo (Including []***NodesUtil**DAOInfo)
2. NodesUtilDAOInfo (Including []*NodeUtilDAOInfo)
3. NodeUtilDAOInfo

get**NodesUtil**iByType() is to find **NodesUtil**DAOInfo of partiticular type in **PartitionNodesUtil**DAOInfo.

PS: I previously struggled with whether to use "Util" or "Utilisation", but in the end, I decided to align with the previous node-utilization API

Reply:
PS: fixed another typo, some abbreviations are incorrect, 'Utili' should be 'Util'.